_adr/87/index.adoc

```diff
@@ -1,48 +1,442 @@
+The description of this issue should have sufficient information on what this new metric should provide in order to identify the metric to be used and how it should be documented in later stages.
+
+The issue created must have the following sub-tasks and they must be completed in order:
+
+. Identify the metric and labels to be used and how it will be aggregated if required.
+** This should be assigned to the team that has the knowledge of the requested metric.
+** [.underline]#Definition of Done#:
+*** Metric to be used is identified.
+*** (Optional) Aggregation of the metric determined.
+*** (Optional) Category/Sub-component which this new metric belongs to is identified.
+*** Cardinality calculated
+*** New metric approved by the Running the Services team to be included in the remote-write to Observatorium.
+
+. Agreement on metric name and labels to be exposed
+** This should be assigned to the Control Plane team.
+** [.underline]#Definition of Done#:
+*** Labels to be exposed are identified.
+*** Name of the metric and labels exposed are determined.
+This should adhere to the metric and label naming policy as described above.
+*** The metric and exposed label names must be approved by the following people/teams:
+**** Requester
+**** Control Plane team
+**** Kafka Integrations team
+
+. Expose the metric in the KAS Fleet Manager API
+** This should be assigned to the Control Plane team.
+** Any tasks for adding the new metric to the API should be done in this stage.
+This includes adding the new metric to the Prometheus remote-write configuration and any code changes to the KAS Fleet Manager repository.
+** [.underline]#Definition of Done#:
+*** New customer facing metric should be included in the response of the following KAS Fleet Manager API endpoints:
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByInstantQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query]
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByRangeQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query_range]
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/federateMetrics[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/federate]
+
+. Document the new metric
+** This should be assigned to the Documentation/Customer Content Support team.
+** [.underline]#Definition of Done#:
+*** New metric documented in the app-services-guides: https://github.com/redhat-developer/app-services-guides/tree/main/docs/kafka/metrics-monitoring-kafka[Monitoring metrics in OpenShift Streams for Apache Kafka].
+
+
+### Migrating existing metrics to the new naming policy
+KAS Fleet Manager does not support API versioning at the moment so we would need to support both existing and new metrics during the migration process.
+Since the revised metric names will be added as new metrics, they will have no history.
```

Comment:
So we're providing the new metrics from the same endpoint as the old metrics? That's a much harder migration story for customers, because they have to check each metric individually whether it's going to be removed. 

I know you're defined this versioning problem as a non-goal, but I wonder whether we should think more about how to solve it, because it's intrinsic to providing a clear migration path. Is there a discussion somewhere about why this is difficult/what the issues are?

Reply:
I believe the underlying issue is that kas fleet manager doesn't really support API versioning at the moment. I believe @wtrocki plans to talk about this in the architecture call :)